src/Surfnet/Stepup/Identity/Identity.php

```diff
@@ -235,24 +238,74 @@ public function verifyEmail($verificationNonce)
         $secondFactorToVerify->verifyEmail();
     }
 
-    public function vetSecondFactor($registrationCode, $secondFactorIdentifier, $documentNumber, $identityVerified)
-    {
-        foreach ($this->verifiedSecondFactors as $secondFactor) {
-            if (!$secondFactor->wouldBeVettedBy(
-                $registrationCode,
-                $secondFactorIdentifier,
-                $documentNumber,
-                $identityVerified
-            )) {
-                continue;
-            }
+    public function vetSecondFactor(
+        IdentityApi $identity,
+        SecondFactorId $secondFactorId,
+        $secondFactorIdentifier,
```

Comment:
I would recommend renaming `$identity` to `$registrant` (indicative of the fact that that identity is the person standing at the desk requesting the vetting). Same with `$secondFactorId` and `$secondFactorType`, it should be clear that those are related to the registrant and not this identity :wink: